app/domain/authentication/security/process_request_proxies.rb

```diff
@@ -0,0 +1,58 @@
+require 'command_class'
+
+module Authentication
+  module Security
+    ProcessRequestProxies ||= CommandClass.new(
+      dependencies: {
+        # InputValidation
+      },
+      inputs: [:request]
+    ) do
+
+      def call
+        validate_request
+        process_xff_header
+      end
+
+      private
+
+      Struct.new('RemoteAddress', :client_ip, :proxy_list)
+
+      def validate_request
+        raise ArgumentError, '@request' if @request.nil?
+        raise ArgumentError, '@request.remote_addr' if nil_or_empty @request.remote_addr
+      end
+
+      def process_xff_header
+        remote_addr = @request.remote_addr
+        xff_header = @request.headers['X-Forwarded-For']
+        xff_header_ip_list = (nil_or_empty xff_header) ? [] : xff_header.to_s.split(/, /)
+        empty_ip_list = xff_header_ip_list.empty?
```

Comment:
this is redundant IMO. you can later call `xff_header_ip_list.empty?` in the `if` clause and it will be clearer.